Meme/Meme/NetworkClient.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  NetworkClient.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/29/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+class NetworkClient {
+
+    fileprivate func getTemplates(_ data: Data?) -> [Template]? {
+        guard let data = data,
+            let jsonResponse = try? JSONSerialization.jsonObject(with: data, options: []) as? [String:Any] else {
+                print("Error deserializing template data");
+                return nil
+        }
+
+        var templates = [Template]()
+        jsonResponse.forEach { (key, value) in
```

Comment:
Should use compactMap here

Reply:
once this is `[String: String]`, you can compactMap (flatMap's new name) like so:
```swift
return jsonResponse.compactMap({ return Template.init(name: $0.0, urlString: $0.1})
```
where that initializer is the failable initializer
The way that compact/flatmap works on optionals is that if the optional is empty aka `.none`, it drops it. For `.some(value)` it returns `value`. So `[.some(1), .none, .some(2)].compactMap ==> [1, 2]`

Reply:
Gotcha, thanks for the explanation!